### PR TITLE
Move address generator from state to environment + some renaming

### PR DESCRIPTION
**Pull request by pattyshack**
1. moved state/address_generator* to environment/account_creator*
2. renamed (g *StateBoundAddressGenerator) to (creator *AccountCreator)
3. renamed stateHolder to stateTransaction
4. refactor Bytes() & constructAddressGen() to use the bytes()
5. also updated some variable names in tests

NOTE: I plan to merge TransactionEnv.CreateAccount (as well as
BootstrapProcedure.createAccount) into AccountCreator as a follow up PR, hence
the renaming.

### CODE REVIEW

File: fvm/environment/account_creator.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const keyAddressState = "account_address_state"
+
+type AccountCreator struct {
```
Comment (Tonix517) [inline fvm/environment/account_creator.go:12]: nit: could you also move over the original comments for StateBoundAddressGenerator?

**Review by Tonix517 — APPROVED** [review]
LGTM. A minor nit

File: fvm/environment/account_creator.go (hunk 1 above)
Comment (pattyshack) [reply fvm/environment/account_creator.go:12]: that comment no longer makes sense once I've merge TransactionEnv.CreateAccount into this.  I'll write some else up

**Review by SaveTheRbtz — APPROVED** [review]

File: fvm/environment/account_creator.go (hunk 1 above)
Comment (pattyshack) [reply fvm/environment/account_creator.go:18]: added comment

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**